specs/rollup-node.md

```diff
@@ -1,167 +1,227 @@
 
-#### Payload attributes derivation
+### Building the Payload Attributes
 
-From the data read from L1, the rollup node constructs an [expanded version of PayloadAttributesV1],
-which includes an additional `transactions` field.
+[payload attributes]: #building-the-payload-attributes
 
-The object properties must be set as follows:
+From the data read from L1 and the encoded transactions, the rollup node constructs the [payload
+attributes][g-payload-attr] as an [expanded version][expanded-paylod] of the [`PayloadAttributesV1`] object, which
+includes an additional `transactions` field.
 
-- `timestamp` is set to the current [unix time],
-  rounded to the closest multiple of 2 seconds. No two blocks may have the same timestamp.
+The object's properties must be set as follows:
+
+- `timestamp` is set to the timestamp of the L1 block.
 - `random` is set to the *random* L1 block attribute
 - `suggestedFeeRecipient` is set to the zero-address for deposit-blocks, since there is no sequencer.
-- `transactions` is an array of the derived deposits, all encoded in the [EIP-2718] format.
+- `transactions` is an array of the derived deposits, encoded as per the two preceding sections.
 
-[expanded version of PayloadAttributesV1]: exec-engine.md#extended-payloadattributesv1
-[unix time]: https://en.wikipedia.org/wiki/Unix_time
-[EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
-[EIP-2930]: https://eips.ethereum.org/EIPS/eip-2930
+[expanded-payload]: exec-engine.md#extended-payloadattributesv1
+[`PayloadAttributesV1`]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#payloadattributesv1
 
-### Output derivation
+------------------------------------------------------------------------------------------------------------------------
 
-Building a full block requires the earlier [derived payload attributes](#payload-attributes-derivation)
-(`payloadAttributes` argument) as well as the previous L2 state (`forkchoiceState` argument), defined by
-the [`engine_forkchoiceUpdatedV1`] method of the [Engine API]:
+## From Payload Attributes to L2 Block
 
-- `headBlockHash`: block hash of the last block of the L2 chain, according to the rollup driver.
-- `safeBlockHash`: same as `headBlockHash`.
-- `finalizedBlockHash`: the hash of the block whose number is `number(headBlockHash) - FINALIZATION_DELAY_BLOCKS` if
-  the number of that block is `>= L2_CHAIN_INCEPTION`, 0 otherwise (where `FINALIZATION_DELAY_BLOCKS == 50400`
-  (approximately 7 days worth of L1 blocks) and `L2_CHAIN_INCEPTION` is the [L2 chain inception] (the number of the
-  first L1 block for which an L2 block was produced). See the [Finalization Guarantees][finalization] section for more
-  details.
+Once the [payload attributes] for a given L1 block `B` have been built, and if we have already derived an L2 block from
+`B`'s parent block, then we can use the payload attributes to derive a new L2 block.
```

Comment:
This 'and if' really stands out to me. What if we haven't?

Reply:
Then we can't use the payload attribute to derive a new L2 block :)
Feel free to suggest a clarification if needed!